src/Stratis.Bitcoin/FinalizedBlockInfoRepository.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                try
+                {
+                    await this.queueUpdatedEvent.WaitAsync(this.cancellation.Token).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                }
+
+                this.queueUpdatedEvent.Reset();
+
+                HashHeightPair lastFinalizedBlock = null;
+
+                lock (this.queueLock)
+                {
+                    while (this.finalizedBlockInfosToSave.Count != 0)
+                        lastFinalizedBlock = this.finalizedBlockInfosToSave.Dequeue();
+                }
+
+                if (lastFinalizedBlock == null)
+                    continue;
+
+                using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
+                {
+                    transaction.Insert<byte[], HashHeightPair>("FinalizedBlock", finalizedBlockKey, lastFinalizedBlock);
+                    transaction.Commit();
+                }
+
+                this.logger.LogTrace("Finalized info saved: '{0}'.", lastFinalizedBlock);
+            }
+
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <inheritdoc />
+        public HashHeightPair GetFinalizedBlockInfo()
+        {
+            return this.finalizedBlockInfo;
+        }
+
+        /// <inheritdoc />
+        public Task LoadFinalizedBlockInfoAsync(Network network)
+        {
+            this.logger.LogTrace("()");
+
+            Task task = Task.Run(() =>
+            {
+                this.logger.LogTrace("()");
+
+                using (DBreeze.Transactions.Transaction transaction = this.dbreeze.GetTransaction())
+                {
+                    transaction.ValuesLazyLoadingIsOn = false;
+
+                    Row<byte[], HashHeightPair> row = transaction.Select<byte[], HashHeightPair>("FinalizedBlock", finalizedBlockKey);
+                    if (!row.Exists)
+                    {
+                        this.finalizedBlockInfo = new HashHeightPair(network.GenesisHash, 0);
+                        this.logger.LogTrace("Finalized block height doesn't exist in the database.");
+                    }
+                    else
+                        this.finalizedBlockInfo = row.Value;
+
+                    this.logger.LogTrace("(-):{0}='{1}'", nameof(this.finalizedBlockInfo), this.finalizedBlockInfo);
+                }
+            });
+
+            this.logger.LogTrace("(-)");
+            return task;
+        }
+
+        /// <inheritdoc />
+        public bool SaveFinalizedBlockHashAndHeight(uint256 hash, int height)
+        {
+            this.logger.LogTrace("({0}:{1})", nameof(height), height);
+
+            if (this.finalizedBlockInfo != null && height <= this.finalizedBlockInfo.Height)
+            {
+                this.logger.LogTrace("(-)[CANT_GO_BACK]:false");
+                return false;
+            }
+
+            var finalizedInfo = new HashHeightPair(hash, height);
```

Comment:
Do we need to create a variable here?

Can't we just use do `this.finalizedBlockInfo = new HashHeightPair(hash, height)` below?

Reply:
Again, this makes complete sense bu maybe just document that?

Reply:
Sure, will do

Reply:
Kk, sure

Reply:
Fixed